packages/api/src/webhooks/cannon.ts

```diff
@@ -123,7 +123,7 @@ export default class WebhookCannon {
   }
 
   async processWebhookEvent(msg: messages.WebhookEvent): Promise<boolean> {
-    const { event, streamId, sessionId, userId } = msg;
+    const { event, streamId, sessionId, userId, projectId } = msg;
```

Comment:
Can you make the `projectId` field non-optional on this `WebhookEvent` type just to have the type safety that we will never send events without a project ID? It is definitely not an optional field, just like the `userId`.

Reply:
In fact, I tried doing that and **several** places broke (most tests, but some actual runtime code [1]). So please do that, fix the errors, and let's keep in mind to use typing in our favor to catch thing like this.

[1] e.g. no projectId passed [here](https://github.com/livepeer/livepeer-com/blob/d983f4573ee3dd47534142eee0430a50551fe6ab/packages/api/src/task/scheduler.ts#L511) :)